### PR TITLE
[NFC] Add NOLINTNEXTLINE for static_cast

### PR DESCRIPTION
**Pull request by MrSidims**
Despite multiple inheritance static_cast from 'SPIRVEntry *' to 'SPIRVFunction *' should be safe. Replacing static_cast with dynamic_cast would be even safer, but it would add extra runtime overhead. So add ignore this clang-tidy check.

### CODE REVIEW

File: lib/SPIRV/libSPIRV/SPIRVInstruction.h [hunk 1]
```diff
@@ -1690,6 +1690,7 @@ class SPIRVFunctionCall : public SPIRVFunctionCallGeneric<OpFunctionCall, 4> {
   SPIRVFunctionCall(SPIRVId TheId, SPIRVFunction *TheFunction,
                     const std::vector<SPIRVWord> &TheArgs, SPIRVBasicBlock *BB);
   SPIRVFunctionCall() : FunctionId(SPIRVID_INVALID) {}
+  // NOLINTNEXTLINE
```
[the remark this answers is not in the document]
Comment (MrSidims) [reply lib/SPIRV/libSPIRV/SPIRVInstruction.h:1693]: Probably yes

**Comment by MrSidims** [comment]
@svenvh please take a look if I'm not doing something wrong here

**Comment by svenvh** [comment]
> @svenvh please take a look if I'm not doing something wrong here

Unfortunate timing I think: the clang-tidy job failed during compile commands generation due to a cmake error.  I hope that should be fixed by llvm-project commit `1c5d12144058 ("[flang] Handle Flang examples consistently with LLVM.", 2023-03-14)` but we need to wait for the apt.llvm.org binaries to catch up.

**Comment by MrSidims** [comment]
>> Unfortunate timing I think

Oh, I meant to ask whether you are ok or not to have such check disabled :)

**Comment by svenvh** [comment]
> Oh, I meant to ask whether you are ok or not to have such check disabled :)

Oh yes, that's fine with me, but I'd like to see its effect on CI before we merge it (to see if it actually silences the issue we were getting all the time).

**Review by asudarsa — APPROVED** [review]
LGTM. Thanks

**Comment by MrSidims** [comment]
Force restart CI to use everything fresh.

**Pull request merged**